src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarView_Partial.cs

```diff
@@ -1645,17 +1645,45 @@ public void SetYearDecadeDisplayDimensions(int columns, int rows)
 			CalendarViewGeneratorHost pHost,
 			DateTime date)
 		{
-			int index = 0;
-
-			index = pHost.CalculateOffsetFromMinDate(date);
+			int index = pHost.CalculateOffsetFromMinDate(date);
 			global::System.Diagnostics.Debug.Assert(index >= 0);
 			global::System.Diagnostics.Debug.Assert(pHost.Panel is { });
+
+#if HAS_UNO_SKIA || __WASM__
```

Comment:
I can't find any usages of `HAS_UNO_SKIA`, what directive do we use for platform specific skia code?

Reply:
We'll need to add tests for this scenario.

Reply:
> I can't find any usages of `HAS_UNO_SKIA`, what directive do we use for platform specific skia code?

`__SKIA__` should work. Though I think this is ported code, so unclear why it needs to change

Reply:
Also note that the current offset looks correct, but the issue is in the "blank" area. It should contain months from the previous year.

The repro steps in the issue showed that it rendered correctly one time, then it no longer renders correctly.


**Correct:**

![image](https://github.com/user-attachments/assets/9de00e7d-c307-47e2-93fd-ed036576fae9)

**Wrong:**

![image](https://github.com/user-attachments/assets/108d68b3-9269-4f45-8f5e-2adaaa3b3483)

Reply:
After digging into this further, I'm not sure what fix/workaround would be best. 

This is the behavior for the UWP CalendarView:

![WindowsCalendarView](https://github.com/user-attachments/assets/b8bccb0b-459e-4e3c-9f90-233bbc7f9b96)

If we compare to the current version @Youssef1313 you are right, the indices are correct it's just the space above that needs to be filled in on other platforms.

Here's where the bounds of the scope of items in the current calendar view are set. If we could decrease the `FirstVisibleIndexBase`, more items would be visible and would fill the gap of this issue.

https://github.com/unoplatform/uno/blob/4ad6ef3bbb9064af827bc2751281cb2e47e31812/src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarView_Partial.cs#L1840-L1841

These indices `FirstVisibleIndexBase` and `LastVisibleIndexBase` are set via code that bubbles up from here:

https://github.com/unoplatform/uno/blob/4ad6ef3bbb9064af827bc2751281cb2e47e31812/src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarLayoutStrategyImpl.cs#L183-L186

I thought I would be able to replicate the changes in this PR in the piece of code above but I've had no luck so far. Everytime we switch from the decade view to the year view the `FirstVisibleIndexBase` increases for some reason I haven't been able to find, leaving even less items visible.

Reply:
@dr1rrb Does that sound familiar?

Reply:
Not really ... I remember that we already did fixes for the month / year / decade switching, but I'm unable to remember what is was, we need to look at git history.